File: CORBin_System/tests/LONG/passlong-client.c

#include <stdio.h>
#include <orb/orbit.h>
#include "passlong.h" 
 
int main (int argc, char *argv[]) {

  passlong  client;      
  CORBA_long val;          
  CORBA_ORB orb=NULL;         /* This is our ORB */
  char *ior;                  /* an IOR is one way a client can locate a server */
  CORBA_Environment ev;       /* CORBA exception information */

  CORBA_exception_init(&ev);

  /* Initialize the orb. The initialization routine checks the command
   * line parameters for directives that affect the orb */
  orb = CORBA_ORB_init(&argc, argv, "orbit-local-orb", &ev);
 
  /* Bind  our client Account to the account on the server using the
   * IOR to look it up (the ior was passed as an argument) */

  client = CORBA_ORB_string_to_object(orb, argv[1], &ev);
  if (!client) {
    printf("Cannot bind to %s\n (pass the IOR as arg 1)\n", argv[1]);
    exit(1);
  }
  val = passlong_foo(client, 187 , &ev);
  printf ("%d was returned from foo.\n",val);

  /* We're done now. Do some cleanup */
  CORBA_Object_release(client, &ev);
  exit (0);
}

File: CORBin_System/tests/LONG/Java/passlong_client.java
import org.omg.CORBA.*;

public class passlong_client {

   public static void main(String args[]) {

	try {
	      //create and init ORB
	      ORB orb = ORB.init(args,null);

	      System.out.println(args[0]);
	      org.omg.CORBA.Object p = orb.string_to_object(args[0]);

	      passlong pl = passlongHelper.narrow(p);

	      System.out.println("Returned: " + pl.foo(198));

	 } catch (Exception e) {

		System.err.println("ERROR: " + e);
		e.printStackTrace(System.out);

	 }	
	      
   }
}


